repo-agent/issue-sandbox/main.go

```diff
@@ -0,0 +1,185 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+)
+
+func main() {
+	cmdCodeSrv, err := startCodeServer()
+	if err != nil {
+		log.Fatalf("failed to start code-server: %v", err)
+	}
+	defer func() {
+		if cmdCodeSrv.Process != nil {
+			_ = cmdCodeSrv.Process.Kill()
+		}
+	}()
+
+	err = runIssueSolver()
+	if err != nil {
+		log.Fatalf("failed solving issue: %v", err)
+	}
+
+	err = cmdCodeSrv.Wait()
+	if err != nil {
```

Comment:
There are many calls to `os.Getenv` scattered throughout the function. It would be cleaner to read all required environment variables at the beginning of the program into a configuration struct. This makes the dependencies on environment variables explicit and easier to manage.